ChartMogul/ChartMogulClient.cs

```diff
@@ -0,0 +1,164 @@
+﻿using ChartMogul.API.Import;
+using OConnors.ChartMogul.API.Models;
+using System;
+using System.Collections.Generic;
+using StructureMap;
+using ChartMogul.API.Models.Core;
+using System.Text;
+using System.IO;
+
+namespace ChartMogul.API
+{
+
+    public interface IChartMogulClient
+    {
+        CustomerModel AddCustomer(CustomerModel customerModel);
+        List<CustomerModel> GetCustomers();
+        void DeleteCustomer(CustomerModel customerModel);
+        List<DataSourceModel> GetDataSources();
+        DataSourceModel AddDataSource(DataSourceModel datasourcemodal);
+        void DeleteDataSource(DataSourceModel dataSourcemodel);
+        PlanModel CreatePlan(PlanModel plan);
+        List<PlanModel> GetPlans();
+        List<InvoiceModel> GetInvoices(CustomerModel customerModel);
+        List<InvoiceModel> AddInvoice(CustomerModel customerModel, List<InvoiceModel> invoiceModellist);
+        TransactionModel AddTransaction(InvoiceModel invoicemodel, TransactionModel transactionmodel);
+        List<SubscriptionModel> GetSubscriptions(CustomerModel customermodelequest);
+        SubscriptionModel CancelSubscription(SubscriptionModel subscriptionModel);
+
+    }
+
+
+    /// <summary>
+    /// Master client for interacting with ChartMogulClient services
+    /// </summary>
+    public class ChartMogulClient : IChartMogulClient
+    {
+        private ICustomer _iCustomer;
+        private IDataSource _iDataSource;
+        private IPlan _iPlan;
+        private IInvoice _iInvoice;
+        private ITransaction _iTransaction;
+        private ISubscription _iSubscription;
+        private APIRequest _apiRequest = new APIRequest();
+
+        public ChartMogulClient(ICustomer iCustomer, IDataSource iDataSource, IPlan iPlan, IInvoice iInvoice, ITransaction iTransaction, ISubscription iSubscription)
+        {
+            _iCustomer = iCustomer;
+            _iDataSource = iDataSource;
+            _iPlan = iPlan;
+            _iInvoice = iInvoice;
+            _iTransaction = iTransaction;
+            _iSubscription = iSubscription;
+        }
+
+
+        public ChartMogulClient()
+        {
+            if (string.IsNullOrEmpty(Configuration.AccountToken))
+            {
+                throw new InvalidDataException("AccountToken cannot be null");
+            }
+
+            if (string.IsNullOrEmpty(Configuration.SecretKey))
```

Comment:
Sure John I got your point .We will change the code to accept account token and secret key from the client constructor code rather than config  values.